queue_services/business-pay/src/business_pay/resources/pay_filer.py

```diff
@@ -0,0 +1,183 @@
+
+    if not (claims := verify_gcp_jwt(request)):
+        return {}, HTTPStatus.FORBIDDEN
+
+    logger.info(f"Incoming raw msg: {str(request.data)}")
+
+    # 1. Get cloud event
+    # ##
+    if not (ce := gcp_queue.get_simple_cloud_event(request, wrapped=True)):
+        #
+        # Decision here is to return a 200,
+        # so the event is removed from the Queue
+        return {}, HTTPStatus.OK
+    logger.info(f"received ce: {str(ce)}")
+
+    # 2. Get payment information
+    # ##
+    if (
+        not (payment_token := get_payment_token(ce))
+        or payment_token.status_code != "COMPLETED"
+    ):
+        # no payment info, or not a payment COMPLETED token, take off Q
+        return {}, HTTPStatus.OK
+
+    # 3. Update model
+    # ##
+    if not (
+        filing := Filing.get_filing_by_payment_token(pay_token=str(payment_token.id))
+    ):
+        # The payment token might not be there yet, put back on Q
+        return {}, HTTPStatus.NOT_FOUND
+
+    if filing.status == Filing.Status.COMPLETED.value:
+        # Already processed, so don't do anything but remove from Q
+        logger.debug(f"already processed, duplicate ce: {str(ce)}")
+        return {}, HTTPStatus.OK
+
+    logger.info(f"processing payment: {payment_token.id}")
+
+    # setting the payment_completion_date, marks the filing as paid
+    filing.payment_completion_date = datetime.now(timezone.utc)
+    filing.payment_status_code = "COMPLETED"
+    filing.status = Filing.Status.PAID
+    filing.save()
+
+    # None of these should bail as the filing has been marked PAID
+    # 4. Publish to email Q
+    # ##
+    with suppress(Exception):
+        mail_topic = current_app.config.get("ENTITY_MAILER_TOPIC", "mailer")
+        email_msg = create_email_msg(filing.id, filing.filing_type)
+        await nats_queue.connect()
+        await nats_queue.publish(subject=mail_topic, msg=email_msg)
+        logger.info(f"publish to emailer for pay-id: {payment_token.id}")
+
+    # 5. Publish to filer Q, if the filing is not a FED (Effective date > now())
+    # ##
+    with suppress(Exception):
+        if filing.effective_date <= filing.payment_completion_date:
+            filer_topic = current_app.config.get("ENTITY_FILER_TOPIC", "filer")
+            queue_message = create_filing_msg(filing.id)
+            await nats_queue.connect()
+            await nats_queue.publish(subject=filer_topic, msg=queue_message)
+            logger.info(f"publish to filer for pay-id: {payment_token.id}")
+
+    logger.info(f"completed ce: {str(ce)}")
+    return {}, HTTPStatus.OK
+
+
+@dataclass
+class PaymentToken:
+    """Payment Token class"""
+
+    id: Optional[str] = None
+    status_code: Optional[str] = None
+    filing_identifier: Optional[str] = None
+    corp_type_code: Optional[str] = None
+
+
+def get_payment_token(ce: SimpleCloudEvent):
+    """Return a PaymentToken if enclosed in the cloud event."""
+    if (
+        (ce.type == "payment")
```

Comment:
This needs to change to https://github.com/bcgov/sbc-common-components/blob/5a8d762cc17b6cc5113f42f775870e52f84e831a/python/src/sbc_common_components/utils/enums.py#L45

Reply:
The message needs to be `bc.registry.payment`